demos/README.md

```diff
@@ -0,0 +1,103 @@
+# Demo apps for the AWS IoT Device SDK for Embedded-C
+
+These programs demonstrate how to use the AWS IoT Device SDK for Embedded-C to
+perform common AWS IoT tasks.
+
+* [HTTP examples](#http-examples)
+* [MQTT examples](#mqtt-examples)
+* [Device shadow examples](#device-shadow-examples)
+
+## HTTP examples
+
+Example directory: `http`
+
+### **http_demo_basic_tls**
+
+This example resolves a domain, establishes a TCP connection, validates the
+server's certificate using the root CA certificate defined in the config header,
+then finally performs a TLS handshake with the HTTP server so that all
+communication is encrypted. After which, HTTP Client library API is used to
+send a GET, HEAD, PUT, and POST request in that order. For each request,
+the HTTP response from the server (or an error code) is logged.
+
+### **http_demo_mutual_auth**
+
+This example resolves the AWS IoT Core endpoint, establishes a TCP connection,
+performs a mutually authenticated TLS handshake occurs such that all further
+communication is encrypted. After which, HTTP Client Library API is used to
+make a POST request to AWS IoT Core in order to publish a message to a topic
+named topic with QoS=1 so that all clients subscribed to the topic receive the
+ message at least once. Any possible errors are also logged.
```

Comment:
This text does not seem correct. It seems to be mixing MQTT information into the HTTP demo.